### PR TITLE
Enable binary operations between scalars and columns of differing decimal types

### LINKED ISSUE #12958: [FEA] Support binary ops between decimal columns and scalars
**Is your feature request related to a problem? Please describe.**
I wish cudf would support binary operations between decimal columns and scalars.

**Describe the solution you'd like**
Currently:
```python
import cudf

df = cudf.DataFrame(
  {
    "a": [1.23, 12.65, 134.64, -34.3, 945.19],
    "b": [1, 1, 2, 2, 3],
  }
)
df["a"] = df["a"].astype(cudf.Decimal64Dtype(10, 2))

df.a < 20
```
or
```python
from decimal import Decimal

df.a < Decimal(20)
```
or
```python
df.a + 20
```
or
```python
df.a * Decimal(2)
```
returns
```
ValueError: CUDF failure at:/opt/conda/conda-bld/work/cpp/src/binaryop/binaryop.cpp:206: Unsupported operator for these types
```

This can be worked around by converting the scalar to a column of the same size as the decimal column and performing the binary operation.

### PR DESCRIPTION
**Pull request by shwina**
Closes #12958 

This PR enables some previously xfailing tests.

### CODE REVIEW

File: python/cudf/cudf/core/column/decimal.py [hunk 1]
```diff
@@ -77,6 +77,9 @@ def _binaryop(self, other: ColumnBinaryOperand, op: str):
             return NotImplemented
         lhs, rhs = (other, self) if reflect else (self, other)
 
+        # breakpoint()
+        # lhs, rhs = self._maybe_cast_up(lhs, rhs)
```
Comment (shwina) [inline python/cudf/cudf/core/column/decimal.py:81]:
```suggestion
        # breakpoint()
        # lhs, rhs = self._maybe_cast_up(lhs, rhs)
```

**Review by shwina — COMMENTED** [review]
Changes

File: python/cudf/cudf/core/column/decimal.py [hunk 2]
```diff
@@ -404,3 +403,7 @@ def _get_decimal_type(lhs_dtype, rhs_dtype, op):
     )
     precision = min(cudf.Decimal128Dtype.MAX_PRECISION, max_precision)
     return cudf.Decimal128Dtype(precision=precision, scale=scale)
+
+
+def _same_precision_and_scale(lhs: DecimalDtype, rhs: DecimalDtype) -> bool:
```
Comment (brandon-b-miller) [inline python/cudf/cudf/core/column/decimal.py:408]: Doesn't `DecimalDtype.__eq__` do this?

Comment (shwina) [reply python/cudf/cudf/core/column/decimal.py:408]: Not when the class of `lhs` and `rhs` are different: (i.e., `cudf.Decimal32Dtype(3, 2) == cudf.Decimal64Dtype(3, 2) == False`)

File: python/cudf/cudf/tests/test_binops.py [hunk 3]
```diff
@@ -3012,10 +3012,10 @@ def decimal_series(input, dtype):
...
 )
 @pytest.mark.parametrize("reflected", [True, False])
-@pytest_xfail(
-    reason="binop operations not supported for different bit-width "
-    "decimal types"
-)
+# @pytest_xfail(
+#     reason="binop operations not supported for different bit-width "
+#     "decimal types"
+# )
```
Comment (brandon-b-miller) [inline python/cudf/cudf/tests/test_binops.py:3018]:
```suggestion
```

**Review by galipremsagar — APPROVED** [review]

**Review by bdice — APPROVED** [review]

**Review by vyasr — APPROVED** [review]

**Pull request merged**